app/stac_api/models.py

```diff
@@ -277,37 +309,66 @@ def __str__(self):
     def clean(self):
         # very simple validation, raises error when geoadmin_variant strings contain special
         # characters or umlaut.
-        if not bool(re.search('^[a-zA-Z0-9]*$', self.geoadmin_variant)):
-            raise ValidationError(_('Property geoadmin:variant not correctly specified.'))
+        try:
+            if not bool(re.search('^[a-zA-Z0-9]*$', self.geoadmin_variant)):
+                raise ValidationError(_('Property geoadmin:variant not correctly specified.'))
+
+        except (KeyError, IndexError) as err:
+            logger.error('Error when validating summaries in assets: %s', err)
+            raise ValidationError('Invalid Summaries')
 
     # alter save-function, so that the corresponding collection of the parent item of the asset
     # is saved, too.
     def save(self, force_insert=False, force_update=False, using=None, update_fields=None):
         self.collection = self.feature.collection
 
-        # check if the collection's geoadmin_variant needs to be updated
-        if self.geoadmin_variant not in self.feature.collection.geoadmin_variant:
-            self.feature.collection.geoadmin_variant.append(self.geoadmin_variant)
-            self.feature.collection.save()
-
-        # proj_epsq (integer) is defined on collection level as well
-        # and eo_gsd (float) on item AND collection level as well.
-        # So we need to check if these properties need an update on parent
-        # and grandparent level.
-        if self.proj_epsq not in self.feature.collection.summaries_proj:
-            self.feature.collection.summaries_proj.append(self.proj_epsq)
-            self.feature.collection.save()
-
-        # for float-comparison:
-        def float_in(flt, floats, **kwargs):
-            return np.any(np.isclose(flt, floats, **kwargs))
-
-        if not float_in(self.eo_gsd, self.feature.collection.summaries_eo_gsd):
-            self.feature.collection.summaries_eo_gsd.append(self.eo_gsd)
-            self.feature.collection.save()
-
-        if not float_in(self.eo_gsd, self.feature.properties_eo_gsd):
-            self.feature.properties_eo_gsd.append(self.eo_gsd)
-            self.feature.collection.save()
+        try:
+            # check if the collection's geoadmin_variant needs to be updated
+            if self.feature.collection.summaries["geoadmin:variant"] is None:
+
+                self.feature.collection.summaries["geoadmin:variant"] = [self.geoadmin_variant]
+                self.feature.collection.save()
```

Comment:
For these checks it would maybe be cleaner to create little helpers/wrappers in the collection model, like `update_variants` or the like. This would keep the internal logic of these fields encapsulated to the collection, item in the end doesn't need to know about how these fields are organized internally. If we change it, it's 'just' a change in the internal representation in the collection and doesn't need to interest others (in the spirit of OOP)

Reply:
Good point, done!